### PR TITLE
Update smb.conf template with additional parameters

### PR DESCRIPTION
**Pull request by anoopcs9**

### CODE REVIEW

File: vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2 [hunk 1]
```diff
@@ -1,19 +1,13 @@
 [global]
+workgroup = MYGROUP
+server string = Samba Server Version %v
+log file = /var/log/samba/log.%m
+max log size = 50
+security = user
+passdb backend = tdbsam
+load printers = yes
```
Comment (obnoxxx) [inline vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2:8]: I don't think we want clustered printing... ;-)

FWIW, we want the options from the original file, that essentially disable printing. Ok, this is done in the include file, but why bother mentioning these options here at all if they are being overwritten anyway?

File: vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2 [hunk 2]
```diff
@@ -1,19 +1,13 @@
 [global]
+workgroup = MYGROUP
+server string = Samba Server Version %v
+log file = /var/log/samba/log.%m
+max log size = 50
```
Comment (obnoxxx) [inline vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2:5]: I don't think want such small logs?

Comment (anoopcs9) [reply vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2]: Let's avoid log rotate by setting it to `0`

File: vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2 (hunk 1 above)
Comment (anoopcs9) [reply vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2]: Ah..my bad.
I am merging both into smb.conf.j2.

File: vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2 [hunk 3]
```diff
@@ -1,19 +1,27 @@
...
-map archive = No
-posix locking = No
+map archive = no
+map hidden = no
+map read only = no
+map system = no
+store dos attributes = yes
 
 load printers = no
-printcap name = /dev/null
```
Comment (obnoxxx) [inline vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2:17]: I think `printcap name = /dev/null` should be kept to completely disable printing:

https://serverfault.com/questions/207510/how-do-you-disable-smb-printing-support
https://lists.samba.org/archive/samba/2006-February/117184.html

Comment (anoopcs9) [reply vagrant/ansible/roles/samba-glusterfs.setup/templates/smb.conf.j2:17]: Done

**Comment by anoopcs9** [comment]
> The diff is a bit bigger than it would need to be due to some re-ordering of lines (and `s/No/no/`...).
Changed in such a way that diff is smaller.

**Pull request merged**